sale_order_type/models/sale_order_type.py

```diff
@@ -33,24 +35,27 @@ def default_picking_policy(self):
     journal_id = fields.Many2one(
         comodel_name="account.journal",
         string="Billing Journal",
-        domain=[("type", "=", "sale")],
+        domain="[('type', '=', 'sale'), '|', ('company_id', '=', False), "
+        "('company_id', '=', company_id)]",
```

Comment:
When the company isn't set and the journal is, this will fail (can a journal have no company?)

Reply:
You are right, the journal company_id field is required, but this domain will be not fails.
If you prefer i can change by [('type', '=', 'sale'), ('company_id', '=', company_id)] ...